### PR TITLE
HDDS-1587. Support dynamically adding delegated classes from to isolated class loader.

### PR DESCRIPTION
**Pull request by chenjunjiedada**

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 69 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | shelldocs | 1 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 67 | Maven dependency ordering for branch |
| +1 | mvninstall | 709 | trunk passed |
| +1 | compile | 390 | trunk passed |
| +1 | checkstyle | 131 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1001 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 215 | trunk passed |
| 0 | spotbugs | 443 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 699 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 29 | Maven dependency ordering for patch |
| +1 | mvninstall | 629 | the patch passed |
| +1 | compile | 416 | the patch passed |
| +1 | javac | 416 | the patch passed |
| +1 | checkstyle | 125 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 26 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 868 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 233 | the patch passed |
| +1 | findbugs | 728 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 251 | hadoop-hdds in the patch failed. |
| -1 | unit | 2318 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 75 | The patch does not generate ASF License warnings. |
| | | 9343 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.common.impl.TestHddsDispatcher |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.TestMiniChaosOzoneCluster |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineProvider |
|   | hadoop.ozone.client.rpc.TestContainerStateMachine |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestBlockDeletion |
|   | hadoop.ozone.client.rpc.TestKeyInputStream |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-942/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/942 |
| Optional Tests | dupname asflicense mvnsite unit shellcheck shelldocs compile javac javadoc mvninstall shadedclient findbugs checkstyle |
| uname | Linux e347765e1b23 4.4.0-144-generic #170~14.04.1-Ubuntu SMP Mon Mar 18 15:02:05 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 101d5b5 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/1/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/1/testReport/ |
| Max. process+thread count | 4375 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozonefs U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/1/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-ozone/common/src/main/bin/ozone-config.sh [hunk 1]
```diff
@@ -49,3 +49,8 @@ else
   exit 1
 fi
 
+# HADOOP_OZONE_DELEGATED_CLASSES defines a list of classes which will be loaded by default
+# class loader of application instead of isolated class loader. With this way we can solve
+# incompatible problem when using hadoop3.x + ozone with older hadoop version.
+#export HADOOP_OZONE_DELEGATED_CLASSES=
```
Comment (xiaoyuyao) [inline hadoop-ozone/common/src/main/bin/ozone-config.sh]: Here we expose a environment variable but the code in FilteredClassLoader only load a system property.

File: hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/FilteredClassLoader.java [hunk 2]
```diff
@@ -57,6 +57,12 @@ public FilteredClassLoader(URL[] urls, ClassLoader parent) {
     delegatedClasses.add("org.apache.hadoop.fs.ozone.OzoneFSStorageStatistics");
     delegatedClasses.add("org.apache.hadoop.fs.ozone.Statistic");
     delegatedClasses.add("org.apache.hadoop.fs.Seekable");
+    String[] dynamicDelegatedClasses =
+        System.getProperty("HADOOP_OZONE_DELEGATED_CLASSES").split(";");
+    for (String delegatedClass : dynamicDelegatedClasses) {
+      delegatedClasses.add(delegatedClass);
+    }
```
Comment (xiaoyuyao) [inline hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/FilteredClassLoader.java]: You can replace the code L60-64 with 
delegatedClasses.addAll(
        getTrimmedStringCollection(System.getenv("HADOOP_OZONE_DELEGATED_CLASSES")));

File: hadoop-ozone/common/src/main/bin/ozone-config.sh (hunk 1 above)
Comment (chenjunjiedada) [reply hadoop-ozone/common/src/main/bin/ozone-config.sh]: I see.

File: hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/FilteredClassLoader.java (hunk 2 above)
Comment (chenjunjiedada) [reply hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/FilteredClassLoader.java]: getTrimmedStringCollection is member function of class Configuration. We can't directly use here. I can add trim() if we care about the space. Is that ok?

Comment (chenjunjiedada) [reply hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/FilteredClassLoader.java]: I see.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 48 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | shelldocs | 1 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 15 | Maven dependency ordering for branch |
| +1 | mvninstall | 567 | trunk passed |
| +1 | compile | 319 | trunk passed |
| +1 | checkstyle | 97 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 862 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 207 | trunk passed |
| 0 | spotbugs | 350 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 555 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for patch |
| +1 | mvninstall | 510 | the patch passed |
| +1 | compile | 357 | the patch passed |
| +1 | javac | 357 | the patch passed |
| -0 | checkstyle | 54 | hadoop-ozone: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 24 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 729 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 192 | the patch passed |
| +1 | findbugs | 599 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 206 | hadoop-hdds in the patch failed. |
| -1 | unit | 1402 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 72 | The patch does not generate ASF License warnings. |
| | | 7205 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.common.impl.TestHddsDispatcher |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestBCSID |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-942/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/942 |
| Optional Tests | dupname asflicense mvnsite unit shellcheck shelldocs compile javac javadoc mvninstall shadedclient findbugs checkstyle |
| uname | Linux e94eb1dffd15 3.13.0-153-generic #203-Ubuntu SMP Thu Jun 14 08:52:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 4ea6c2f |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/2/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/2/testReport/ |
| Max. process+thread count | 5290 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozonefs U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/2/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 41 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | shelldocs | 1 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 15 | Maven dependency ordering for branch |
| +1 | mvninstall | 527 | trunk passed |
| +1 | compile | 319 | trunk passed |
| +1 | checkstyle | 101 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 858 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 202 | trunk passed |
| 0 | spotbugs | 409 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 636 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 28 | Maven dependency ordering for patch |
| +1 | mvninstall | 494 | the patch passed |
| +1 | compile | 326 | the patch passed |
| +1 | javac | 326 | the patch passed |
| -0 | checkstyle | 56 | hadoop-ozone: The patch generated 7 new + 0 unchanged - 0 fixed = 7 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 25 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 747 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 186 | the patch passed |
| +1 | findbugs | 568 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 207 | hadoop-hdds in the patch failed. |
| -1 | unit | 1593 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 68 | The patch does not generate ASF License warnings. |
| | | 7363 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.common.impl.TestHddsDispatcher |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestBCSID |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineProvider |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-942/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/942 |
| Optional Tests | dupname asflicense mvnsite unit shellcheck shelldocs compile javac javadoc mvninstall shadedclient findbugs checkstyle |
| uname | Linux 7e65e91a685e 3.13.0-153-generic #203-Ubuntu SMP Thu Jun 14 08:52:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 4ea6c2f |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/3/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/3/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/3/testReport/ |
| Max. process+thread count | 5201 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozonefs U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/3/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 32 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | shelldocs | 1 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 32 | Maven dependency ordering for branch |
| +1 | mvninstall | 503 | trunk passed |
| +1 | compile | 318 | trunk passed |
| +1 | checkstyle | 102 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 812 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 198 | trunk passed |
| 0 | spotbugs | 338 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 534 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 30 | Maven dependency ordering for patch |
| +1 | mvninstall | 470 | the patch passed |
| +1 | compile | 291 | the patch passed |
| +1 | javac | 291 | the patch passed |
| -0 | checkstyle | 57 | hadoop-ozone: The patch generated 7 new + 0 unchanged - 0 fixed = 7 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 23 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 696 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 196 | the patch passed |
| +1 | findbugs | 553 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 181 | hadoop-hdds in the patch failed. |
| -1 | unit | 1383 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 69 | The patch does not generate ASF License warnings. |
| | | 6872 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.common.impl.TestHddsDispatcher |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-942/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/942 |
| Optional Tests | dupname asflicense mvnsite unit shellcheck shelldocs compile javac javadoc mvninstall shadedclient findbugs checkstyle |
| uname | Linux e5559dd49ca1 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 23c0379 |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/4/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/4/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/4/testReport/ |
| Max. process+thread count | 4590 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozonefs U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/4/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 30 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 16 | Maven dependency ordering for branch |
| +1 | mvninstall | 494 | trunk passed |
| +1 | compile | 313 | trunk passed |
| +1 | checkstyle | 96 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 809 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 196 | trunk passed |
| 0 | spotbugs | 340 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 532 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 30 | Maven dependency ordering for patch |
| +1 | mvninstall | 477 | the patch passed |
| +1 | compile | 326 | the patch passed |
| +1 | javac | 326 | the patch passed |
| +1 | checkstyle | 112 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 20 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 712 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 193 | the patch passed |
| +1 | findbugs | 553 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 159 | hadoop-hdds in the patch failed. |
| -1 | unit | 1184 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 70 | The patch does not generate ASF License warnings. |
| | | 6660 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.common.impl.TestHddsDispatcher |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-942/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/942 |
| Optional Tests | dupname asflicense mvnsite unit shellcheck shelldocs compile javac javadoc mvninstall shadedclient findbugs checkstyle |
| uname | Linux 3e7e69b5588c 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 23c0379 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/5/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/5/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/5/testReport/ |
| Max. process+thread count | 5396 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozonefs U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-942/5/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by xiaoyuyao** [comment]
The unit test failures are not related to this change. +1 and I will merge it shortly.

**Pull request merged**